docs/src/examples/lcs/lcs.jl

```diff
@@ -0,0 +1,305 @@
+# & LCS(X*c_1, Y*c_2) \in \left\{ LCS(X*c_1, Y), LCS(X, Y*c_2)\right\} & \text{if } c_1 \neq c_2
+# \end{align}
+# ```
+#
+# where $X*c$ denotes the concatenation of sequence $X$ with character $c$. The
+# first property above allows simplifying the Longest Common Subsequence
+# computation of two sequences that end with the same character. When the two
+# sequences end with different characters, the second property allows
+# simplifying the problem to a choice between two possibilities.
+#
+# If we now denote by $L_{i,j}$ the length of the longest common subsequence
+# between the first $i$ characters of $X$ and the first $j$ characters of $Y$,
+# then we now have the following property for $L$:
+#
+# ```math
+# L_{i,j} = \left\{
+# \begin{array}{ll}
+# 0 & \text{ if } i=0 \text{ or } j=0\\
+# 1 + L_{i-1, j-1} & \text{ if } i\neq 0 \text{ and } j\neq 0 \text{ and } x_i = y_j\\
+# \text{max}\left(L_{i-1, j}, L_{i,j-1}\right) & \text{ if } i\neq 0 \text{ and } j\neq 0 \text{ and } x_i \neq y_j\\
+# \end{array}
+# \right.
+# ```
+#
+# where $x_i$ represents the $i$-th character in sequence $X$ and $y_j$ the $j$-th character in sequence $Y$.
+#
+# If $L$ is stored in an array, the first condition above allows initializing
+# its first row and first column. Then, the neighboring cells can be filled
+# using the two other conditions, which in turn unlocks further neighboring
+# cells until the whole $L$ matrix is filled.
+
+# ## Small example
+#
+# Let us begin with a small example: we want to find a longest common subsequence between `GAC` and `AGCAT`:
+
+x = collect("GAC");
+y = collect("AGCAT");
+
+# Let's build the $L$ array. For the sake of the example, we'll initially fill
+# it with $(-1)$ in order to indicate which values haven't been computed yet. The
+# `init_length!` function fills the first row and the first column in the
+# array. We also define a `display` helper function, which shows a pretty
+# representation of the $L$ array.
+
+function init_lengths!(L, x, y)
+    @inbounds L[1,1] = 0
+    @inbounds for i in eachindex(x)
+        L[i+1, 1] = 0
+    end
+    @inbounds for j in eachindex(y)
+        L[1, j+1] = 0
+    end
+end
+
+using PrettyTables
+display(L, x, y) = pretty_table(hcat(['∅', x...], L);
+                                header = [' ', '∅', y...],
+                                formatters = (v, i, j) -> v==-1 ? "" : v)
+
+L = -ones(Int, 1+length(x), 1+length(y))
+init_lengths!(L, x, y)
+display(L, x, y)
+
+# The `fill_lengths!` function then allows filling other rows. By default it
+# will fill the entire array, but for the sake of the example we restrict it
+# here to a subset of the rows and/or the columns.
+
+function fill_lengths!(L, x, y, ir=eachindex(x), jr=eachindex(y))
+    @inbounds for j in jr
+        for i in ir
+            L[i+1, j+1] = if x[i] == y[j]
+                L[i, j] + 1
+            else
+                max(L[i+1, j], L[i, j+1])
+            end
+        end
```

Comment:
Is the one liner obscure?
```suggestion
        for i in ir
            L[i+1, j+1] = x[i] == y[j] ? L[i, j] + 1 : max(L[i+1, j], L[i, j+1])
        end
```

Reply:
May I suggest also merging the loops?
```
for j in jr, i in ir
    ...
end
```

Reply:
Yes, good ideas!